pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java

```diff
@@ -169,6 +181,27 @@ private BrokerResponse handleRequest(long requestId, String query,
       return new BrokerResponseNative(QueryException.getException(QueryException.SQL_PARSING_ERROR, e));
     }
 
+    QueryPlan queryPlan = queryPlanResult.getQueryPlan();
+    Set<String> tableNames = getTableNamesFromRelRoot(queryPlanResult.getRelRoot());
```

Comment:
we have the exact same util in pinot-controller that listed out all the accessed tables --> then find the appropriate broker to route the query to.

however that's based off SqlNode instead of RelNode. I was thinking about unifying these two later. we can keep a TODO github issue for reminder if we decided to merge this PR as is

Reply:
CC @ankitsultana @tibrewalpratik17

Reply:
Created https://github.com/apache/pinot/issues/10576